src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -1278,6 +1279,76 @@ public void RegisterMissingIncludes(BlittableJsonReaderArray results, BlittableJ
             }
         }
 
+        internal void RegisterCounters(BlittableJsonReaderArray counters, bool gotAll, string[] ids)
+        {
+            if (CountersByDocId == null)
+            {
+                CountersByDocId = new Dictionary<string, (bool GotAll, Dictionary<string, long?> Values)>(StringComparer.OrdinalIgnoreCase);
+            }
+
+            if (counters == null || counters.Length == 0)
+            {
+                if (gotAll == false || ids.Length == 0)
+                    return;
+                
+                foreach (var id in ids)
```

Comment:
Why do you need to create an empty dictionary here?

Reply:
Same here, if I have no counters in result but asked to get all, then I want to remember that and not go to server again, no?